### PR TITLE
Add wifi provisioning BLE service

### PR DESCRIPTION
**Pull request by wentong-li**
Extract common part from wifi provisioning sample, and form it as a BLE service.

### CODE REVIEW

**Review by alwa-nordic — APPROVED** [review]

**Comment by wentong-li** [comment]
Hi @richabp . The doc is reorganized as follows:

`readme.rst`: previously it is a navigation page containing two subpages. Now it holds the contents of `sample_description.rst` and no longer has subpage. **Now it is for the sample.**
`sample_description.rst`: removed
`implementaion_details.rst`: it is moved to doc/nrf/libraries/bluetooth_services/services, and renamed as `wifi_prov.rst`. **It is for the service.**

Hope my explanation can help.

File: doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst [hunk 1]
```diff
@@ -1,102 +1,25 @@
...
-
-.. _wifi_provisioning_message_types:
-
-Message types
-=============
-
-The sample uses four message types:
-
-* ``Info``: Allows the configurator to know the capability of a device before the provisioning process happens.
+This module implements of an Bluetooth GATT service for Wi-Fi provisioning.
```
Comment (kittydepa) [inline doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:13]:
```suggestion
This module implements a Bluetooth® GATT service for Wi-Fi® provisioning.
```

File: doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst [hunk 2]
```diff
@@ -1,102 +1,25 @@
...
-
-See :ref:`wifi_provisioning_message_types` for more information on the commands and additional parameters.
-
-Message delivery
-****************
-
-Exchanging messages requires a secure transport method.
-The Wi-Fi Provisioning Service sample uses Bluetooth® Low Energy (LE).
-You can assign different security levels to components under one service to fine-tune access control.
+* GATT service inerface: Defines GATT service and provide an interface for configurator to exchange messages.
```
Comment (kittydepa) [inline doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:17]:
```suggestion
* GATT service inerface: Defines GATT service and provides an interface for the configurator to exchange messages.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst [hunk 3]
```diff
@@ -155,42 +78,100 @@ The purpose of each characteristic is as follows:
...
-The advertising interval depends on the provisioning status.
-If the device is not provisioned, the interval is 100 ms. If it is provisioned, the interval is 1 second.
+See :ref:`wifi_provisioning_message_types` for more information on the commands and additional parameters.
 
 Configuration management
 ************************
 
 The configuration management component manages Wi-Fi configurations.
-It uses the wifi_credentials library to handle the configurations in flash.
+It uses the :ref:`lib_wifi_credentials` to handle the configurations in flash.
```
Comment (richabp) [inline doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:164]:
```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```

Comment (richabp) [inline doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:164]:
```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```

**Review by richabp — CHANGES_REQUESTED** [review]
Please add an entry for the updated Provisioning sample in the change log.

**Review by kittydepa — COMMENTED** [review]
Added one article, within Richa's suggested edit.

File: samples/wifi/provisioning/README.rst [hunk 4]
```diff
@@ -7,13 +7,133 @@ Wi-Fi: Provisioning Service
...
+The sample is divided into three parts:
+
+* Task and event handling component: Handles provisioning-related tasks and events.
+* Transport layer: Based on Bluetooth Low Energy and exchanges information between the device and the nRF Wi-Fi Provisioner app.
+* Configuration management component: Manages the provisioning data (in RAM and flash) accessed by multiple threads.
+
+Features
+********
+
+Kconfig options
```
[the remark this answers is not in the document]
Comment (wentong-li) [reply samples/wifi/provisioning/README.rst:37]: I will add a section called "configuration" to list all kconfig options we provide, and explain their functions

**Review by bama-nordic — APPROVED** [review]
LGTM

File: samples/wifi/provisioning/README.rst [hunk 5]
```diff
@@ -7,13 +7,137 @@ Wi-Fi: Provisioning Service
...
+
+The sample requires a smartphone (configurator) with Nordic Semiconductor's nRF Wi-Fi Provisioner app installed in one of the following versions:
+
+* `nRF Wi-Fi Provisioner mobile app for Android`_
+* `nRF Wi-Fi Provisioner mobile app for iOS`_
+
+Overview
+********
+
+With this sample, you can provision a Wi-Fi® device that lacks input or output capability using the :ref:`wifi_prov_readme` library.
```
Comment (kittydepa) [inline samples/wifi/provisioning/README.rst:29]:
```suggestion
With this sample, you can provision a Wi-Fi device that lacks input or output capability using the :ref:`wifi_prov_readme` library.
```

File: samples/wifi/provisioning/README.rst [hunk 6]
```diff
@@ -7,13 +7,137 @@ Wi-Fi: Provisioning Service
...
+The sample is divided into three parts:
+
+* Task and event handling component: Handles provisioning-related tasks and events.
+* Transport layer: Based on Bluetooth® Low Energy and exchanges information between the device and the nRF Wi-Fi Provisioner app.
```
Comment (kittydepa) [inline samples/wifi/provisioning/README.rst:33]:
```suggestion
* Transport layer: Based on Bluetooth Low Energy and exchanges information between the device and the nRF Wi-Fi Provisioner app.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]
Two small changes, to remove trademark symbols since they have already been used in the first instance.

**Review by kittydepa — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**